velox/functions/lib/DateTimeFormatter.h

```diff
@@ -173,9 +175,9 @@ class DateTimeFormatter {
     return tokens_;
   }
 
-  // If failOnError is false, returns std::nullopt for parsing error.
-  // Otherwise, fail with error thrown.
-  std::optional<DateTimeResult> parse(
+  // If failOnError is false, returns Unexpected with Invalid status for
+  // parsing error. Otherwise, returns Unexpected with UserError status.
+  Expected<DateTimeResult> parse(
```

Comment:
I assume we no longer need failOnError parameter. We can simply change the function to not throw and report errors via 'Unexpected'. We can further include clear error message so that the caller doesn't need to provide a message of their own (unless they want to add context that's not available in this function).